### PR TITLE
Low Mach-number Approximate Riemann Solver option (LMARS) 

### PR DESCRIPTION
**Pull request by akshaysridhar**
# Description

Adds additional option, LMARS to the available list of interface numerical fluxes. 
1) Tested against Riemann problem [1D]
2) Tested against Rising Thermal Bubble [2D]
3) Tested against Rising Thermal Bubble [3D]

# TODO 
- [x] Consolidate test cases with @slifer50 , based on MoistRoe flux option
- [x] Confirm boundary condition interface changes / requirements
- [x] Add docstrings, rebase 

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and run `julia .dev/climaformat.jl .`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Review by Zhengyu-Huang — COMMENTED** [review]
Do we need to have a linearized version for IMEX?

**Comment by akshaysridhar** [comment]
bors try

**Comment by akshaysridhar** [comment]
bors try

**Comment by akshaysridhar** [comment]
bors try

**Comment by szy21** [comment]
Is this ready to go? Could someone take a look?

**Review by Zhengyu-Huang — APPROVED** [review]
I think these tests are enough, thanks for your efforts!

File: src/Atmos/Model/linear.jl [hunk 1]
```diff
@@ -510,3 +510,30 @@ function numerical_flux_first_order!(
...
+    state_prognostic⁻::Vars{S},
+    state_auxiliary⁻::Vars{A},
+    state_prognostic⁺::Vars{S},
+    state_auxiliary⁺::Vars{A},
+    t,
+    direction,
+) where {S, A}
+
+    numerical_flux_first_order!(
+        RusanovNumericalFlux(),
```
Comment (simonbyrne) [inline src/Atmos/Model/linear.jl:528]: Is Rusanov a good choice here? It doesn't satisfy the contact property.

Comment (szy21) [reply src/Atmos/Model/linear.jl:528]: @Zhengyu-Huang @akshaysridhar Is Rusanov ok? Or would central be better?

Comment (akshaysridhar) [reply src/Atmos/Model/linear.jl:528]: @simonbyrne Linear-acoustics results in the Rusanov expression for this flux method
